### DISCUSSION TITLE
Is MAUI Blazor on Web Support Adsense Ads in the Future?

### DISCUSSION THREAD

**Discussion by nextcodelab** [dotnet/maui#3179] [General]
I always have plan to create a websites and monetize using Adsense ads, like any other popular websites across entire globe.
There  are popular websites using adsense, CNN, BBC, The VERGE, ABS, Csharp Corner, USA TODAY, Fox News , MSN  and thousands to millions of websites want to monitize there sites or use adsense.. IF MAUI BLAZOR ON WEB DOES NOT SUPPORT ADSENSE IN THE FUTURE, HOW THE CREATOR OF THAT SITES EARN IF THEY ONLY RELY ON ADS/VISITORS? IT SEEMS THAT THIS BLAZOR IS USELESS TO OTHER WEBSITES CREATORS THAT RELY ON ADS EARNING. 

Sorry for my brief message, I cannot constructs sentences in english fluently.

**Comment by GeraudFabien** [comment]
Hello,
Maui doesn't work on web so i assume when you say maui blazor you talk about blazor. 

In blazor you can call any JavaScript library (It may need more or less work depending on the goal and how much it will interact with the dom and you're app). But you shouldn't have problem with adsense (has far i know). If you do have then it should be you're problem you write here.

### **Anyway this thread should be on the aspnetcore repo with blazor.**
If you mean in maui then i don't understand you're last sentence.
> IT SEEMS THAT THIS BLAZOR IS USELESS TO OTHER WEBSITES CREATORS.

Other than who ? You ? But has i understand you mean should have write it useless to me. There is a lot of enterprise application. And most single page application (SPA) are enterprise because this way of distributing work badly with SEO even with all the work done to improve this part. But it work very well in enterprise by diminishing the server load after the first contact and splitting the app.

Also :
> Sorry for my brief message, I cannot constructs sentences in english fluently.

If you where truly sorry you :
- Wouldn't have use uppercase.
- Write blazor or MAUI instead of maui blazor we can't even know what you talking about
- Any web dev know of adsense. You don't have to explain what it is and list site you think used id

**Reply by nextcodelab** [reply]
Thank you so much for your reply,,, I am not javascript dev and maybe for me it takes years learning javascript, css and html to build a websites to use adsense. I just hoping that someday I can create site using blazor web and earn by putting adsense on it someday.
There is also video in youtube that the adsense ads is working on blazor, but this video is outdated from current and future blazor web version or in maui blazor.

2020
[https://www.youtube.com/watch?v=GjLEVlUA7mY](https://www.youtube.com/watch?v=GjLEVlUA7mY)

2021
[https://www.youtube.com/watch?v=cMHWZWjkEtM](https://www.youtube.com/watch?v=cMHWZWjkEtM)

Website that created in Blazor that have adsense ads.
[https://fairplaytube.pticostarica.com/Public/Videos/Details/aa38759d7c](https://fairplaytube.pticostarica.com/Public/Videos/Details/aa38759d7c)

MSN Uses Adsense
[https://www.msn.com/](https://www.msn.com)

**Comment by nextcodelab** [comment]
[UPDATE]
I can now be able to insert adsense script on razor web or maui blazor web .
```
<div @key="@Guid.NewGuid()">
    @if (Ad != null)
    {
        @Ad
    }
</div>

@code {
    private RenderFragment Ad { get; set; }
   
    protected override void OnInitialized()
    {
        Ad = new RenderFragment(b =>
        {
            b.OpenElement(0, "ins");
            b.AddMultipleAttributes(1, new List<KeyValuePair<string, object>>()
            {
                new KeyValuePair<string, object>("class", "adsbygoogle"),
                new KeyValuePair<string, object>("style","display:block"),
                #if DEBUG
                new KeyValuePair<string, object>("data-adtest", "on"),
                #endif
                new KeyValuePair<string, object>("data-ad-client", "ca-pub-9882496897289_fake"),
                new KeyValuePair<string, object>("data-ad-slot", "1086038_fake"),
                new KeyValuePair<string, object>("data-ad-format", "auto"),
                new KeyValuePair<string, object>("data-full-width-responsive",true)
            });
            b.CloseElement();

            b.OpenElement(2, "script");
            b.AddContent(3, "(adsbygoogle = window.adsbygoogle || []).push({});");
            b.CloseElement();
        });
    }
}

```

**Reply by wind2006** [reply]
How to pass the adsense audit without a website?
